message-broker/rabbitmq/README.md

```diff
@@ -0,0 +1,307 @@
+## RabbitMQ
+
+RabbitMQ is a message broker that supports multiple messaging protocols.
+
+In this example, we will generate a Singularity container running RabbitMQ 3.7.8 for Alpine Linux 3.8.
+This container also includes the installation of Python 3.7.1 to follow the example on sending and receiving a message to the queue.
+
+
+#### What you need:
+
+To run this example you will need:
+
+ - Singularity, which you can download and install from [here](https://github.com/sylabs/singularity)
+ - A text editor (e.g. `vim` or `nano`)
+ - Root access
+
+#### Sections in the definition file:
+
+We'll go through the explanation of every section in the definition file:
+
+In the `%help` section, you can find the description for the container with the respective version of RabbitMQ, Python (needed to run the example) and Alpine.
+
+In the `%environment` section, the environment variables needed are defined.
+
+In the `%post` section, the needed dependencies are installed and also the exported variables are defined. Basically all the needed dependencies from Erlang and Python are installed in order to run the example consistently.
+
+After this, on the `%startscript` section, whenever a container instance is started, the server is started and the status is printed on the output of the console. By default, the server will start listening on node `localhost` on port `5672`.
+
+#### Setting up the environment:
+
+To run this example, we will work on a folder called `rabbitmq` located at the home directory. You can create it with the next command like so:
+
+```
+$ mkdir ~/rabbitmq
+$ cd ~/rabbitmq
+```
+
+After this, we will need the next folders inside  `~/rabbitmq`:
+
+```
+$ mkdir -p ~/rabbitmq/var/{lib/rabbitmq,log}
+```
+
+Then, please do make sure you have the following directory structure before following the example:
+
+ ```
+ ~/rabbitmq
+        |---var
+             |--lib
+             |    |--rabbitmq
+             |--log
+```
+
```

Comment:
Maybe offer to pull the container:

```
$ singularity pull library://sylabs/examples/rabbitmq:latest
```
I'll push & sign the container when your ready.

Reply:
Opps! nevermined!

Reply:
I forget you had a pull command below. Sorry :/